src/index.js

```diff
@@ -2,9 +2,21 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
+import { Provider } from 'react-redux'; // redux bindings for react
+import thunk from 'redux-thunk'; // to use promises for asynchronous actions
+import { createStore, applyMiddleware, compose } from 'redux'; // to create the store and middleware
+import reducers from './reducers/index.js';
 import * as serviceWorker from './serviceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const middleware = [thunk];
+const store = createStore(reducers, {}, compose(applyMiddleware(...middleware)));
```

Comment:
One note, Let's extract this `{}` into it's own JS file called initialStore.js.

What we're aiming to do is to have this one initialStore.js manage an initial state so if we add parameters to the store, we add it in this file rather than here. We'll export the initialStore.js' main object and import it here for `createStore`

Reply:
Also: We may want to add the redux devtools integration here.
```
 const store = createStore(
   reducer, /* preloadedState, */
+  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
 );
```
https://github.com/zalmoxisus/redux-devtools-extension#11-basic-store